### ISSUE TITLE
Question: Set active tab to original scene on stack on tapping a tab item

### ISSUE THREAD

**Issue by arunmenon1975** [grahammendick/navigation#325]
I think this qualifies more as a question than an issue since i may perhaps be missing some prop or setting.

TLDR;

a) press a tabitem and navigate to the corresponding screen 
b) Press an inner button here to navigate to another screen (non-tabbed perhaps)
c) Press any tabitem(could even be the same item tapped earlier) to get back to the main tab scene for that tab item
d) presented with the screen from b) while i expect to see the screen from a) 

I have this in my App file:

```
const stateNavigator = new StateNavigator([
  { key: 'mainmenu', trackCrumbTrail: true },
  { key: 'maindailyspecials', trackCrumbTrail: true },
  { key: 'mainmore', trackCrumbTrail: true },
  { key: 'mainadmin', trackCrumbTrail: true },
]);
 
const { mainmenu, maindailyspecials, mainmore, mainadmin } = stateNavigator.states;
mainmenu.renderScene = () => <MainMenu />;
maindailyspecials.renderScene = () => <MainDailySpecials />;
mainmore.renderScene = () => <MainMore />;
mainadmin.renderScene = () => <MainAdmin />;

const dailySpecialsNavigator = new StateNavigator(stateNavigator);
const mainMoreNavigator = new StateNavigator(stateNavigator);

stateNavigator.navigate('mainmenu');
dailySpecialsNavigator.navigate('maindailyspecials');
mainMoreNavigator.navigate('mainmore');

     <TabBar bottomTabs={true}>
        <TabBarItem title="Menu">
          <NavigationHandler stateNavigator={stateNavigator}>
            <NavigationStack
              primary={true}
              crumbStyle={from => from ? 'scale_in' : 'scale_out'}
              unmountStyle={from => from ? 'slide_in' : 'slide_out'} />
          </NavigationHandler>
        </TabBarItem>
        <TabBarItem
          title="Daily Specials"
          onPress={() => {
          }
          }
        >
          <NavigationHandler stateNavigator={dailySpecialsNavigator}>
            <NavigationStack
              primary={false}
              crumbStyle={from => from ? 'scale_in' : 'scale_out'}
              unmountStyle={from => from ? 'slide_in' : 'slide_out'} />
          </NavigationHandler>
        </TabBarItem>
        .....
        .... more tab items
        .....
        <TabBarItem
          title="More..."

          onPress={() => { }}>
          <NavigationHandler stateNavigator={mainMoreNavigator}>
            <NavigationStack
              primary={false}
              crumbStyle={from => from ? 'scale_in' : 'scale_out'}
              unmountStyle={from => from ? 'slide_in' : 'slide_out'} />
          </NavigationHandler>
        </TabBarItem>
      </TabBar>


```
I navigate to the `More...` tab which is basically a screen with a list of further links. An `Admin` link here will navigate to the Admin screen. I now tap on any tab item to move away from the Admin screen. I then proceed to tap on the `More` tab item again. I am now presented with the `Admin` screen while i am expecting to see the `More` screen. 

My More file:

```
const MainMore: React.FunctionComponent = () => {
    const [submitting, setSubmitting] = useState(false);
    const { stateNavigator } = useContext(NavigationContext)
    const goToAdmin = () => {
        stateNavigator.navigate("mainadmin")
    }
    return (<>
        <View style={{ flex: 1, flexDirection: "column", }}>
            <Text>
                More
                </Text>
            <View style={{ flexDirection: "column", flexGrow: 1, flexShrink: 1, flexBasis: "auto" }}>
                <ActionButton title="To Admin" submitting={submitting} onPress={goToAdmin} type="tertiary" />
            </View>

        </View>
    </>
    )
}
export default MainMore;
```

How do i reset the screen to `More` on tapping the tab after i've move away from it earlier? I can of course navigate anywhere using buttons within the Admin screen(or even the back buttons/links), but this issue only surfaces on attempting via taps on tab items.

**Comment by grahammendick** [comment]
Hey, good to hear from you. Let's take the Twitter sample to see if I've understood you correctly. 

You tap the notifications tab to see the list of notifications. You tap on a notification to go to that user's profile. Then you tap the Home tab. Are you saying that you if tap the notifications tab again you'd expect to go back to the list of notifications instead of staying on the user's profile?

**Comment by arunmenon1975** [comment]
Thanks for the recent updates to the library. Both the cross-platform `TabBar` and `NavigationBar` updates are core components and i depend on them quite a bit and have just started my integration using them. 

wrt my question, yes, thats exactly what i was referring to. 

Im not sure if this is a platform limitation/feature however though, from a UX perspective, it just doesn't sound correct.

**Comment by mpiannucci** [comment]
That’s expected behavior on iOS because UINavigationController retains its state within the tab controller

**Comment by grahammendick** [comment]
Yes, the Android implementation is a copy of the native iOS implementation.
@arunmenon1975 Are you expecting the Tab Bar to behave differently on Android and iOS?

**Comment by arunmenon1975** [comment]
> That’s expected behavior on iOS because UINavigationController retains its state within the tab controller

Ok. Interestingly, there is iOS related information (besides Android) on the Material site describing the expected [behaviour for Bottom Navigation](https://material.io/components/bottom-navigation/#behavior) which goes to say the same:

> Bottom navigation behaves differently on Android and iOS. When you select a bottom navigation item (one that’s not currently selected), each platform displays different outcomes:
> 
> On Android: the app navigates to a destination’s top-level screen. Any prior user interactions and temporary screen states are reset, such as scroll position, tab selection, and in-line search.
> On iOS: the destination reflects the user’s prior interaction. If the user previously visited that section of the app, they return to the last screen viewed (with its prior state preserved, if possible). Otherwise, the app navigates to the top-level screen.
>

**Comment by arunmenon1975** [comment]
> Are you expecting the Tab Bar to behave differently on Android and iOS?

I guess from a consistency perspective in cross-platform solutions keeping interactions similar makes sense. i.e. from the app user perspective, who gets to see the same predictable flow regardless of device OS. But this may not hold good from  a platform perspective, if the platform (and thus the user with expectations) expects following strictures.

In my own app, my current thought process is that i am looking for native predictable consistency as much as possible (thus may not have, for example, a hamburger menu in IOS). This does limit my possibilities quite a bit and my app does look like it will end up 'drabbish' and predictable but importantly will start to look authentically native. Getting the best of both worlds does involve sacrifices, unfortunately.

I think that going forward, and by my above argument, i have 2 options:

1) my 5th and last tab titled `More...` will open a scene with a listing of further links in iOS while on Android i can have it open a DrawerLayout menu (which can also be reached via a dedicated hamburger menu). A Navigation Drawer is also the [suggested approach](https://material.io/components/bottom-navigation/#usage) in the Material guidelines:

> Don’t.
> 
> Don’t use more than five destinations. For those cases, try tabs or a navigation drawer

though i guess they meant to not use the bottom tabs at all and go full-in with top-tabs or drawer navigation. Its in a grey area of just about passing the guideline so it should be ok in solutions where cross-platform consistency is trying to be adhered to.

2) or go Drawer Navigation all the way in Android. Top Tabs may still not work since i think scrolling tabitems is not currently possible in the new `TabBar` component  and i may have a need for a few more top tabitems than can be easily set to a fixed width tab bar.

I can easily decide between any of these two options.

**Comment by grahammendick** [comment]
I understand that you might want a different tab implementation on both platforms. But if you change the way the tabs behave then you need a completely different design on both platforms. What I mean is that if the Tab Bar remembers navigation state on iOS but not on Android then it wouldn't work as a cross platform component. Does that make sense?

You can create your own Bottom Navigation out of `Buttons` and a single `NavigationStack`. Then onPress of each button you navigate to the initial scene of the tab. As long as you set `trackCrumbTrail` to false (the default) on each of these start states then the stack will begin from scratch each time. 
```jsx
<View style={flex: 1}>
  <NavigationHandler stateNavigator={stateNavigator}>
    <NavigationStack />
  </NavigationHandler>
</View>
<View style={height: 76}>
  <Button onPress= />
  <Button onPress= />
</View>
```

> scrolling tabitems is not currently possible

I'm happy if you want to work on a PR to add scrolling tabitems. Let me know if you're interested.

**Comment by arunmenon1975** [comment]
> I'm happy if you want to work on a PR to add scrolling tabitems. Let me know if you're interested.

Thanks for the offer. Will take a look at the code and will confirm in a separate comment. I just want to make sure that things are not too complex for me before committing.

> But if you change the way the tabs behave then you need to design your app very differently on both platforms. 

Yes, and i was trying to avoid this (dual designs) as much as possible. But i suspect it may eventually be an inevitability as the app features increase over time and that i stick to my original plan and have things as naive as possible on both platforms. This is one of the reasons i am taking time to finalize the IA and deciding which components to use where. My best bet i think is to assume there will be a hive-off point in the future where i may need separate files per platform (and thus designs, even if the design changes may be minimal) i.e. `file.android.js` and a `file.ios.js` and pro-actively create them from the beginning.

The provided example is definitely interesting and viable. In fact i had something like this as my first attempt, pre-integration of the Navigation router. I think i will go this route on Android since i will only have at max 5 items that will be displayed in the bottom navigation on both platforms. 

Reg Android, the Material guidelines do mention to exercise caution when evaluating using both `bottom navigation` and `top tabs`, but in my case there is reason to have both. In fact the [example clips in the Navigation ](https://material.io/components/bottom-navigation/#behavior)section of the docs have both (and a header as well, incidentally) so it is ok to pursue this approach. Now if i could only get the scrolling tab items going.

**Comment by grahammendick** [comment]
Making the tabs scrollable is a good PR for you. The work is similar to your "badge props" PR but this time it's on Android instead of iOS

**Comment by arunmenon1975** [comment]
Will send the PR in a short while. I think i know how to get a minimal scrollable tab going.

**Comment by arunmenon1975** [comment]
Please see #328

**Comment by grahammendick** [comment]
Are you happy to close this now that I've merged in #328? Thanks again for the PR

**Comment by arunmenon1975** [comment]
I have a related issue. I hope its ok to continue the discussion here before closing since there is pertinent content related to the question and the discussions.

I will add the issue over 2 comment blocks since i also have a 'distracting' screen clip that i want to add for better clarity. A comment block just for the image will allow me to hide it and link to it on demand.

**Comment by arunmenon1975** [comment]
Screen clip with issue:

![sc-rec-tabs](https://user-images.githubusercontent.com/2016484/68761480-25c8ae00-063a-11ea-8386-d30ea6509198.gif)

The relevant code for this follows in the next comment.

**Comment by arunmenon1975** [comment]
> You can create your own Bottom Navigation out of Buttons and a single NavigationStack. Then onPress of each button you navigate to the initial scene of the tab. As long as you set trackCrumbTrail to false (the default) on each of these start states then the stack will begin from scratch each time.
> ```
>   <View style={flex: 1}>
>     <NavigationHandler stateNavigator={stateNavigator}>
>       <NavigationStack />
>     </NavigationHandler>
>   </View>
>   <View style={height: 48}>
>     <Button onPress= />
>     <Button onPress= />
>   </View>
> ```



I went about attempting the bottom tabs approach but have an issue when i try and add a component that has the tabs integrated as top tabs. 

Bottom Tab 1(BT 1) is the default scene that is navigated to on launch. This will open up a Top Tabs scene inside the `View` above the bottom tabs section. This has the TabBar integrated with 2 tabs. It works fine on launch. But when i navigate away to another scene via the bottom tabs i get a black screen. The NavigationBar remains intact however. When i attempt navigation betwen bottom tabs that dont have inner `TabBar` components, things work fine. I think [the screen clip in the above comment](https://github.com/grahammendick/navigation/issues/325#issuecomment-553371053) will allow for further clarity.

Please see my relevant code below.

App.tsx:

```
import React, { useState } from 'react';
import { Button, Text, View, TouchableOpacity, StyleSheet, ViewStyle, TextStyle, Touchable, TouchableOpacityProps } from 'react-native';
import { StateNavigator } from 'navigation';
import { NavigationHandler } from 'navigation-react';
import { NavigationBar, NavigationStack, TitleBar } from 'navigation-react-native';

import BottomTab1 from './BottomTab1';

interface IScreenProps {
  title?: string;
}

interface IStyles {
  mainTabContentContainer: ViewStyle;
  navBarTitleBar: ViewStyle;
  navBarTitle: TextStyle;
  bottomTabBar: ViewStyle;
  bottomTabButton: TouchableOpacityProps;
}

type TGoToMainTab = (id: string, title: string) => void;

const BottomTab2:React.FunctionComponent = () => {
  return <View style={{ flex: 1 }}>
    <Text> Bottom Tab 2 content </Text>
  </View>
}
const BottomTab3 :React.FunctionComponent= () => {
  return <View style={{ flex: 1 }}>
    <Text> Bottom Tab 3 content </Text>
  </View>
}
const BottomTab4:React.FunctionComponent = () => {
  return <View style={{ flex: 1 }}>
    <Text> Bottom Tab 4 content </Text>
  </View>
}
const BottomTab5:React.FunctionComponent = () => {
  return <View style={{ flex: 1 }}>
    <Text> Bottom Tab 5 content</Text>
  </View>
}


const stateNavigator = new StateNavigator([
  { key: 'bottomTab1' },
  { key: 'bottomTab2' },
  { key: 'bottomTab3' },
  { key: 'bottomTab4' },
  { key: 'bottomTab5' },
]);
const { bottomTab1, bottomTab2, bottomTab3, bottomTab4, bottomTab5 } = stateNavigator.states;
bottomTab1.renderScene = () => <BottomTab1 />;
bottomTab2.renderScene = () => <BottomTab2 />;
bottomTab3.renderScene = () => <BottomTab3 />;
bottomTab4.renderScene = () => <BottomTab4 />;
bottomTab5.renderScene = () => <BottomTab5 />;

stateNavigator.navigate('bottomTab2');
stateNavigator.navigate('bottomTab1');

const App: React.FunctionComponent<IScreenProps> = (props) => {
  const [title, setTitle] = useState(props.title);
  const goToMainTab: TGoToMainTab = (id, title) => {
    setTitle(title);
    stateNavigator.navigate(id);
  };
  return (
    <>
      <View style={appStyles.mainTabContentContainer}>
        <NavigationBar>
          <TitleBar style={appStyles.navBarTitleBar}>
            <Text style={appStyles.navBarTitle}> {title} </Text>
          </TitleBar>
        </NavigationBar>

        <NavigationHandler stateNavigator={stateNavigator}>
          <NavigationStack />
        </NavigationHandler>
      </View>
      <View style={appStyles.bottomTabBar}>
        <TouchableOpacity style={appStyles.bottomTabButton} onPress={() => goToMainTab("bottomTab1", "Bottom Tab 1")}>
          <Text> BT 1 </Text>
        </TouchableOpacity>
        <TouchableOpacity style={appStyles.bottomTabButton} onPress={() => goToMainTab("bottomTab2", "Bottom Tab 2")}>
          <Text> BT 2 </Text>
        </TouchableOpacity>
        <TouchableOpacity style={appStyles.bottomTabButton} onPress={() => goToMainTab("bottomTab3", "Bottom Tab 3")}>
          <Text> BT 3 </Text>
        </TouchableOpacity>
        <TouchableOpacity style={appStyles.bottomTabButton} onPress={() => goToMainTab("bottomTab4", "Bottom Tab 4")}>
          <Text> BT 4 </Text>
        </TouchableOpacity>
        <TouchableOpacity style={appStyles.bottomTabButton} onPress={() => goToMainTab("bottomTab5", "Bottom Tab 5")}>
          <Text> BT 5 </Text>
        </TouchableOpacity>
      </View>

    </>
  );
}
App.defaultProps = {
  title: "Bottom Tab 1"
};


const appStyles = StyleSheet.create<IStyles>({
  mainTabContentContainer: {
    flex: 1
  },
  navBarTitleBar: {
    flex: 1,
    alignItems: "flex-start",
    justifyContent: "center"
  },
  navBarTitle: {
    fontSize: 18
  },
  bottomTabBar: {
    flexBasis: "auto",
    flexGrow: 0,
    flexShrink: 1,
    flexDirection: "row",
    height: 48,
    justifyContent: "space-evenly",
    backgroundColor: "lightblue",
  },
  bottomTabButton: {
    backgroundColor: "blue",
    padding: 10,
    borderRadius: 10,
    alignSelf: "stretch",
    flex: 1,
    margin: 2,
    justifyContent: "center",
    alignItems: "center"
  },
})

export default App;

```


BottomTabs1.tsx:

```
import { StateNavigator } from 'navigation';
import { NavigationHandler } from 'navigation-react';
import { NavigationStack, TabBar, TabBarItem } from 'navigation-react-native';
import React, { useState } from 'react';
import { Text, View } from 'react-native';

const TopTab1 = () => {
  return <View style={{ flex: 1 }}>
    <Text> Top Tab 1 content </Text>
  </View>
}
const TopTab2 = () => {
  return <View style={{ flex: 1 }}>
    <Text> Top Tab 2 content </Text>
  </View>
}


const stateNavigatorTopTab1_1 = new StateNavigator([
  { key: 'topTab1', },
  { key: 'topTab2', trackCrumbTrail: true },

]);

const { topTab1, topTab2 } = stateNavigatorTopTab1_1.states;
topTab1.renderScene = () => <TopTab1 />;
topTab2.renderScene = () => <TopTab2 />;

const stateNavigatorTopTab1_2 = new StateNavigator(stateNavigatorTopTab1_1);

stateNavigatorTopTab1_1.navigate('topTab1');
stateNavigatorTopTab1_2.navigate('topTab2');

interface IScreenProps {
}

const BottomTab1: React.FunctionComponent<IScreenProps> = (props) => {
  return (
    <>
      <TabBar bottomTabs={false} selectedTintColor="red">
        <TabBarItem  title="Top Tab 1">
          <NavigationHandler stateNavigator={stateNavigatorTopTab1_1}>
            <NavigationStack
              primary={true}
              crumbStyle={from => from ? 'scale_in' : 'scale_out'}
              unmountStyle={from => from ? 'slide_in' : 'slide_out'} />
          </NavigationHandler>
        </TabBarItem>
        <TabBarItem
          title="Top Tab 2"
          onPress={() => {
          }
          }
        >
          <NavigationHandler stateNavigator={stateNavigatorTopTab1_2}>
            <NavigationStack
              primary={false}
              crumbStyle={from => from ? 'scale_in' : 'scale_out'}
              unmountStyle={from => from ? 'slide_in' : 'slide_out'} />
          </NavigationHandler>
        </TabBarItem>
      </TabBar>
    </>
  );
}
export default BottomTab1;

```

 Im not sure why thats happening. Please let me know if there is some issue with the way i am attempting the integration.

**Comment by grahammendick** [comment]
Try setting `primary={false}` on both the inner tab stacks.

**Comment by arunmenon1975** [comment]
That indeed works. Thank you. 

The very last question i have here is the black screen that appears while transitioning between the bottom tab scenes. I tried removing the `NavigationBar` and as well as set an explicit  background colour to the enclosing `View`:

`mainTabContentContainer: {
    flex: 1,
    backgroundColor: "white"
  },`

but it still remains.

**Comment by grahammendick** [comment]
You should turn off the mounting animation for the first scenes of each tab
```jsx
bottomTab1.unmountStyle = 'none';
bottomTab2.unmountStyle = 'none';
...
<NavigationHandler stateNavigator={stateNavigator}>
  <NavigationStack unmountStyle={(from, state) => state.unmountStyle} />
</NavigationHandler>
```

**Comment by arunmenon1975** [comment]
Yes, that fixes the issue as well. I really haven't looked deeply into `unmountStyle`/`crumbStyle` and a lot of other features as yet.

Thank again.
